projects/core/components/button/button.component.ts

```diff
@@ -110,7 +110,7 @@ export class TuiButtonComponent
     }
 
     get loaderSize(): TuiSizeS {
-        return this.size === 'l' ? 'm' : 's';
+        return this.size === 'l' || this.size === 'xl' ? 'm' : 's';
```

Comment:
We have also `sizeBigger` function in /core package for size comparisons, but it is not so important to use this function :)

Reply:
Ah, good to know. Next time :-)